examples/cosmos/ics20/denomTrace.qnt

```diff
@@ -173,6 +188,87 @@ module denomTraceTest {
   }))
 }
 
+module denomTraceSimulation {
+  import denomTrace.*
+
+  pure val chains = Set("A", "B", "C")
+  pure val transferSequence = List(
+    ("A", "B"),
+    ("B", "C"),
+    ("C", "A"),
+    ("A", "C"),
+    ("C", "B"),
+    ("B", "A")
+  )
+
+  const channelPredicate: (str -> str -> str) => bool
+
+  pure val channelNames = Set("channel-1", "channel-2", "channel-3")
+
+  /// All possible combination of channels by chain satisfying `channelPredicate`
+  pure val possibleChannelSetups = chains.setOfMaps(chains.setOfMaps(channelNames)).filter(channelPredicate)
+
+  var trace: DenomTrace
+  var channelSetup: str -> str -> str
```

Comment:
I had to make this a variable in order to pick one value from the `possiblChannelSetups` and make it fixed for the entire run. If you can think of a better way, lmk.

Reply:
Yeah, currently, it is the best way to do it. Perhaps, we could add something similar to `--cinit` of Apalache